### PR TITLE
Use nix wrappers where possible.

### PR DESCRIPTION
**Pull request by qwandor**
This removes a bunch of unsafe blocks and error-checking code.

Also added tests for `local_addr` and `peer_addr` methods.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -487,7 +412,7 @@ impl IntoRawFd for VsockStream {
 
 impl Drop for VsockStream {
     fn drop(&mut self) {
-        unsafe { close(self.socket) };
+        _ = close(self.socket);
```
Comment (Tim-Zhang) [inline src/lib.rs:415]: `let _` is better, same reason with above

File: src/lib.rs [hunk 2]
```diff
@@ -207,7 +199,7 @@ impl IntoRawFd for VsockListener {
 
 impl Drop for VsockListener {
     fn drop(&mut self) {
-        unsafe { close(self.socket) };
+        _ = close(self.socket);
```
Comment (Tim-Zhang) [inline src/lib.rs:202]: This syntax is unstable on lower rust versions like 1.52, I think let _ is better in compatibility

File: tests/vsock.rs [hunk 3]
```diff
@@ -71,3 +76,25 @@ fn test_vsock() {
...
+    assert_eq!(local_addr.port(), LISTEN_PORT);
+}
+
+#[test]
+fn test_stream_addresses() {
+    let stream =
+        VsockStream::connect(&VsockAddr::new(SERVER_CID, SERVER_PORT)).expect("connection failed");
+
+    let local_addr = stream.local_addr().unwrap();
+    assert_eq!(local_addr.cid(), libc::VMADDR_CID_ANY);
```
Comment (Tim-Zhang) [inline tests/vsock.rs:95]: Got error: 
```
---- test_stream_addresses stdout ----
thread 'test_stream_addresses' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `4294967295`', tests/vsock.rs:95:5
```

Comment (qwandor) [reply tests/vsock.rs]: Weird, it passes for me. What OS/version/architecture are you testing on?

In any case, I've made the test accept either result.

File: src/lib.rs (hunk 2 above)
Comment (qwandor) [reply src/lib.rs]: Done.

File: src/lib.rs (hunk 1 above)
Comment (qwandor) [reply src/lib.rs]: Done.

File: tests/vsock.rs (hunk 3 above)
Comment (Tim-Zhang) [reply tests/vsock.rs]: It works now.

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @qwandor

**Pull request merged**